sql/hive/src/main/scala/org/apache/spark/sql/hive/TableReader.scala

```diff
@@ -176,12 +176,13 @@ class HadoopTableReader(
               val matches = fs.globStatus(pathPattern)
               matches.foreach(fileStatus => existPathSet += fileStatus.getPath.toString)
             }
-            // convert  /demo/data/year/month/day  to  /demo/data/*/*/*/
+            // convert  /demo/data/year/month/day  to  /demo/data/year/month/*/
```

Comment:
This is a pretty old logic. Can you explain what's going on here and why your change works? It can help other people to understand your change quickly,

Reply:
how about `select * from test where A='00'`? Shall we check one more level?

Reply:
and more trickily, `select * from test where B='01'`?

Reply:
Em... It seems we have to check all the levels unless we have specified a value for each partition column. We can make some improvement here but seems that require more complicated approach.

Reply:
If the partition columns are `A/B/C/D`, unless you specify a value for `A`, you have to check that level. The same case for `B`/`C`/`D`

Reply:
@cloud-fan @jiangxb1987
Thanks a lot for review.
> Em... It seems we have to check all the levels unless we have specified a value for each partition column. We can make some improvement here but seems that require more complicated approach.
Yes, true. In this change, I only optimize when user specify for each partition column, which is very common in the production -- as our user always did: `select xxx from yyy where year=yy and month=mm and day=dd`

I'm not sure about you guys idea:  leave the current logic as it is(at least the code logic now is very simple)? or implement a more complicated approach and defend as many cases as possible? or do some improvement based on this pr and cover some very common cases?

Thanks again for review :)

Reply:
Actually do we still need this check? now we have something like `spark.sql.files.ignoreMissingFiles`

Reply:
can we make `spark.sql.files.ignoreMissingFiles` work for hive scan? it seems a better solution than this check.

Reply:
Sure, that would be great. Is there some existing pr/jira working on that? if not, I can make the change :)